scripts/common.py

```diff
@@ -781,14 +781,18 @@ def managed_process(command_args, shell=False, timeout_secs=60, **kwargs):
         yield popen_proc
     finally:
         if popen_proc.is_running():
-            procs = popen_proc.children(recursive=True) + [popen_proc]
-            for proc in procs:
-                proc.terminate()
-
-            _, still_running = psutil.wait_procs(procs, timeout=timeout_secs)
-            for proc in still_running:
-                proc.kill()
-                logging.warn('Process killed (pid=%d)' % proc.pid)
+            procs_to_kill = popen_proc.children(recursive=True) + [popen_proc]
+            survivors = []
+            for proc in procs_to_kill:
+                if proc.is_running():
+                    survivors.append(proc)
+                    proc.terminate()
```

Comment:
I'm assuming this function to be solely responsible for starting and terminating the process, but I think now we are assuming that the process can be terminated by another function/code. If it's just for testing then should we add a log on which process where found already terminated? (with the proc.name?)

Reply:
That's a great idea! I've added some debug information and unit tests to make sure they work as expected.